### PR TITLE
support datadiff meta filter

### PR DESCRIPTION
**Pull request by dlawin**
Adds support for the meta filter described here:
https://docs.datafold.com/guides/dbt_advanced_configs#filter-tables

The filter expression is applied to both sides of the diff

Resolves partially: #518

### CODE REVIEW

File: data_diff/dbt.py [hunk 1]
```diff
@@ -191,7 +192,16 @@ def _get_diff_vars(
         dev_qualified_list = [dev_database, dev_schema, model.alias]
         prod_qualified_list = [prod_database, prod_schema, model.alias]
 
-    return DiffVars(dev_qualified_list, prod_qualified_list, primary_keys, dbt_parser.connection, dbt_parser.threads)
+    where = None
+    if model.meta:
+        try:
+            where = model.meta["datafold"]["datadiff"]["filter"]
```
Comment (dlawin) [inline data_diff/dbt.py:198]: very open to suggestion here -- not very pretty

File: data_diff/cloud/datafold_api.py [hunk 2]
```diff
@@ -101,6 +101,8 @@ class TCloudApiDataDiff(pydantic.BaseModel):
     table1: List[str]
     table2: List[str]
     pk_columns: List[str]
+    filter1: Optional[str]
```
Comment (pik94) [inline data_diff/cloud/datafold_api.py:104]:
```suggestion
    filter1: Optional[str] = None
    filter2: Optional[str] = None
```

File: data_diff/dbt.py [hunk 3]
```diff
@@ -73,6 +73,7 @@ class DiffVars:
     primary_keys: List[str]
     connection: Dict[str, str]
     threads: Optional[int]
+    where: Optional[str]
```
Comment (pik94) [inline data_diff/dbt.py:76]:
```suggestion
    where: Optional[str] = None
```

File: data_diff/dbt.py [hunk 4]
```diff
@@ -191,7 +192,16 @@ def _get_diff_vars(
         dev_qualified_list = [dev_database, dev_schema, model.alias]
         prod_qualified_list = [prod_database, prod_schema, model.alias]
 
-    return DiffVars(dev_qualified_list, prod_qualified_list, primary_keys, dbt_parser.connection, dbt_parser.threads)
+    where = None
```
Comment (pik94) [inline data_diff/dbt.py:195]: Maybe `where_filter` or `where_clause` are better because more readable.

File: data_diff/dbt.py (hunk 1 above)
Comment (pik94) [reply data_diff/dbt.py:198]: **nit**: For this or future PRs.

A datafold datadiff section might contain other useful different parameters such as include/exclude_columns, time_travel columns etc which we may want to use in the future for running cloud diffs.

I think it is worth to have a special classmethod of `DbtParser` or separate function to extract from a model that section if it exists.
```
           # dbt_parser.py
           class TDatafoldDatadiffModelParams(pydantic.BasicModel):
               filter_: Optional[str] = None
               include_columns: Optional[List[str]] = None
               exclude_columns: Optional[List[str]] = None

            class DbtParser:  
                  @classmethod
                  def get_datafold_datadiff_params(cls, model) -> TDatafoldDatadiffModelParams:
                      # implementation
            
            # dbt.py     
            datafold_datadiff_params = dbt_parser.get_datafold_datadiff_params()
            return DiffVars(
                 dev_qualified_list, prod_qualified_list, primary_keys, dbt_parser.connection, dbt_parser.threads, datafold_datadiff_params.filter_)
```

File: data_diff/dbt.py (hunk 4 above)
Comment (dlawin) [reply data_diff/dbt.py]: done

File: data_diff/dbt.py (hunk 1 above)
Comment (dlawin) [reply data_diff/dbt.py]: good point. I'm about to add include/exclude columns and I will refactor in that PR

**Review by vvkh — APPROVED** [review]

**Pull request merged**